utils/neural_magic/utils.py

```diff
@@ -0,0 +1,68 @@
+from re import search
+from typing import Any, Dict
+
+import torch
+from sparsezoo import Model
+
+from utils.torch_utils import ModelEMA
+
+__all__ = ["sparsezoo_download", "ToggleableModelEMA", "load_ema"]
+
+
+class ToggleableModelEMA(ModelEMA):
+    """
+    Subclasses YoloV5 ModelEMA to enabled disabling during QAT
+    """
+
+    def __init__(self, enabled, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.enabled = enabled
+
+    def update(self, *args, **kwargs):
+        if self.enabled:
+            super().update(*args, **kwargs)
+
+
+def sparsezoo_download(path: str) -> str:
+    """
+    Loads model from the SparseZoo and override the path with the new download path
+    """
+    model = Model(path)
+    file = _get_model_framework_file(model, path)
+    path = file.path
+    return path
+
+
+def _get_model_framework_file(model: torch.nn.Module, path: str) -> str:
```

Comment:
isn't something like this in sparseml already? download_framework_files_by_recipe_type maybe?

Reply:
Good point! Updated, but pending landing of neuralmagic/sparseml#1140